lib/motions/open-file-under-cursor.coffee

```diff
@@ -0,0 +1,58 @@
+{Motion} = require './general-motions'
+path = require 'path'
+{Directory} = require 'atom'
+shell = require 'shell'
+{exec} = require 'child_process'
+utils = require '../utils'
+
+class OpenFileUnderCursor extends Motion
+  execute: (count) ->
+    ###
+      /usr/bin/test
+      ../../.travis.yml
+      ./search-motion
+      ./index
+      http://github.com
+      https://thing.com/item%20test
+      //github.com
+    ###
+    wordRegex = /[a-z0-9\.\-_\/\\%:]+/i
+    @editor.getCursors().forEach (cursor) =>
+      currentWord = utils.getCurrentWord @editor, cursor, wordRegex
+      # range = cursor.getCurrentWordBufferRange wordRegex: wordRegex
+      # selectedPath = @editor.getTextInRange range
+      range = currentWord.range
+      selectedPath = currentWord.word
+
+      # exit early for url match
+      if /^https?:/i.test(selectedPath)
+        return @openUrl selectedPath
+      if /^\/\//i.test(selectedPath)
+        return @openUrl "http:#{selectedPath}"
+
+      if selectedPath[0] isnt '/' and selectedPath[0] isnt '\\'
+        selectedPath = path.join path.dirname(@editor.buffer.file.path), selectedPath
+
```

Comment:
There are some ex functions that work on files like this but I can't think of anything else that would be in vim-mode.  So, yea, doesn't make sense from a code reusability standpoint.

I was just thinking from a readability standpoint: seems like the first half of the function is doing a pretty different job from the second half.  The parts might be easier to understand than the whole.

But, like before, I might not be seeing the whole picture here.  :)